### PR TITLE
Added a way to auto update av definitions

### PR DESCRIPTION
**Pull request by pballandras**
Runs a simple timer in a separate thread and calls an update every 24 hours.

Improvement paths: Set an hour for the start of the timer (so that it runs more like a cron), hook this into the CLI to customize timer.

Relates to DT-4019

### CODE REVIEW

File: cvdupdate/auto_updater.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+from cvdupdate.cvdupdate import CVDUpdate
+
+# one time per day
+WAIT_TIME_SECONDS = 24 * 60 * 60
+
+
+def start() -> None:
+    """Spawns a thread to update the AV db periodically"""
+    Thread(target=_update, daemon=True).start()
```
Comment (pballandras) [inline cvdupdate/auto_updater.py]: `daemon=True` is for the thread to die quietly when `main` is killed

File: cvdupdate/__main__.py [hunk 2]
```diff
@@ -209,6 +209,7 @@ def serve(port: int, config: str, verbose: bool):
     m = CVDUpdate(config=config, verbose=verbose)
     os.chdir(str(m.db_dir))
     m.logger.info(f"Serving up {m.db_dir} on localhost:{port}...")
+    auto_updater.start()
```
Comment (dblanchette) [inline cvdupdate/__main__.py]: That changes the default behaviour, which reduces the chance of being accepted to the upstream repo.

I suggest an option `--update-interval-seconds` with the default being 0 (feature disabled)

File: cvdupdate/auto_updater.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+def _update() -> None:
+    """Don't call this directly
+    Updates the AV db every day when it was started"""
+    ticker = Event()
+    m = CVDUpdate()
+    while not ticker.wait(WAIT_TIME_SECONDS):
+        errors = m.db_update(debug_mode=True)
+        if errors > 0:
+            m.logger.error(f"Failed to fetch updates from ClamAV databases")
```
Comment (dblanchette) [inline cvdupdate/auto_updater.py]: Useless `f`. Flake8 recognizes this as an error, for example.

Comment (pballandras) [reply cvdupdate/auto_updater.py]: woops I had just copy pasted a log statement didn't see that

File: cvdupdate/__main__.py [hunk 4]
```diff
@@ -200,16 +200,17 @@ def clean_all(config: str, verbose: bool):
 @cli.command("serve")
 @click.option("--config", "-c", type=click.Path(), required=False, default="", help="Config path. [optional]")
 @click.option("--verbose", "-V", is_flag=True, default=False, help="Verbose output. [optional]")
+@click.option("--update-interval-seconds", "-u",type=click.INT, required=False, default=0, help="Time in seconds before the next database update")
```
Comment (dblanchette) [inline cvdupdate/__main__.py]: - Shorthand is usually in capital (`-U`)
- (OCD) Missing space before `type=click.INT`

File: cvdupdate/__main__.py [hunk 5]
```diff
@@ -200,16 +200,17 @@ def clean_all(config: str, verbose: bool):
...
 @click.argument("port", type=int, required=False, default=8000)
-def serve(port: int, config: str, verbose: bool):
+def serve(port: int, config: str, verbose: bool, update_interval_seconds: int):
```
Comment (dblanchette) [inline cvdupdate/__main__.py:205]: I doubt mypy runs on this, but a return paramter type hint would be welcome

File: cvdupdate/auto_updater.py [hunk 6]
```diff
@@ -2,21 +2,23 @@
 
 from cvdupdate.cvdupdate import CVDUpdate
 
-# one time per day
-WAIT_TIME_SECONDS = 24 * 60 * 60
+def start(interval: int) -> None:
+    '''Spawns a thread to update the AV db after "interval" seconds
```
Comment (dblanchette) [inline cvdupdate/auto_updater.py]: - OCD: Please use `"""` and not `'''`
- OCD again: `Spawns` -> `Spawn` https://stackoverflow.com/a/59902236

**Review by dblanchette — APPROVED** [review]

**Pull request merged**